plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryClientFactory.java

```diff
@@ -54,11 +76,15 @@ public BigQueryClientFactory(
             IdentityCacheMapping identityCacheMapping,
             BigQueryCredentialsSupplier credentialsSupplier,
             BigQueryConfig bigQueryConfig,
+            Optional<BigQueryProxyConfig> bigQueryProxyConfig,
```

Comment:
Can we extract some entity to configure transport instead of passing optional binding here?

Reply:
Yes. I refactored this PR and noticed along the way that gRPC requests were not proxied. It was harder to do but I've managed and will be pushing shortly.